### PR TITLE
Feature/network request

### PR DESCRIPTION
**Pull request by davidleach724**
# Description
This feature incorporates retrieving data from a web API. It pulls the data to display all movies as well as retrieving data for a single movie based on that movie's id. If there is an error while fetching the data, a new state property is created and that error is displayed on the page. 

## Type of change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactor Code

## How Has This Been Tested?
- [x] open index.html
- [x] console.log()
- [x] dev tools

## Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas

## Additional Comments for Team Members if Applicable 
This PR successfully pulls the data, but the displayed data still needs to be refactored and cleaned up on a new branch.

### CODE REVIEW

**Review by Samantha-Brown — APPROVED** [review]
The dynamic fetch call and error catch is very clean and efficient. Great working with you!

File: src/APIcalls.js [hunk 1]
```diff
@@ -0,0 +1,4 @@
+export const fetchData = (dataSet) => {
+  return fetch(`https://rancid-tomatillos.herokuapp.com/api/v2/${dataSet}`)
+  .then(response => response.json())
+}
```
Comment (sertmer) [inline src/APIcalls.js:4]: Nice! don't foget to include a catch for the errors

File: src/App.js [hunk 2]
```diff
@@ -1,20 +1,27 @@
...
 import Movies from './Movies'
 import SingleMovie from './SingleMovie'
+import { fetchData } from './APIcalls'
 
 class App extends Component {
   constructor() {
     super()
     this.state = {
-      movies: { movieData },
+      movies: '',
```
Comment (sertmer) [inline src/App.js:10]: Movies is going to be an array eventaully, right? Then it's default value should be an array too!

File: src/App.js [hunk 3]
```diff
@@ -1,20 +1,27 @@
...
       singleMovie: ''
```
Comment (sertmer) [inline src/App.js:11]: Same goes for single movie, only I'd imagine its going to be an object eventually

File: src/App.js [hunk 4]
```diff
@@ -1,20 +1,27 @@
...
+  componentDidMount() {
+    fetchData('movies')
+    .then(movies => this.setState({ movies }))
+    .catch(error => this.setState({ error: error}))
+  }
+
   chooseMovie = (id) => {
-    const chosenMovie = this.state.movies.movieData.movies.find(movie => movie.id === id)
-    this.setState({ singleMovie: chosenMovie })
+    fetchData(`movies/${id}`)
```
Comment (sertmer) [inline src/App.js:22]: Love the dynamic fetch!

File: src/SingleMovie.js [hunk 5]
```diff
@@ -12,14 +11,14 @@ const SingleMovie = ({ singleMovie, clearMovie }) => {
...
         <div className="right-side">
-          <h4 className="movie-overview">"Some overview that is full of buzzwords to attempt to entice you to watch this movie! Explosions! Drama! True love! Robots! A cute dog!"</h4>
-          <h5>"It's a movie!"</h5>
-          <p>Release Date: </p>
-          <p>Run Time: </p>
-          <p>Genres: </p>
-          <p>Avg. Rating: </p>
-          <p>Budget: </p>
-          <p>Revenue: </p>
+          <h4 className="movie-overview">{singleMovie.overview}</h4>
```
Comment (sertmer) [inline src/SingleMovie.js:14]: Good opportunity to destructure the keys out of the singleMove object, so you don't have to type it again and again

**Review by sertmer — COMMENTED** [review]
Nice work! Check the comments for some refactoring ideas!

**Pull request merged**